### PR TITLE
sqllogictest: A logging and command line filter

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

 re https://github.com/apache/arrow-datafusion/issues/4460

# Rationale for this change

While debugging a sqllogictest I would like to run a single file at a time rather than all in the directory. I would also like the ability to see debug logs, etc, etc

# What changes are included in this PR?

1. Initialize `env_logger` so setting `RUST_LOG` is respted
2. Add ability to specify files on the command line,
3. Documentation

Note, I did not bring in Clap or some other full featured command line parser. I figured if/when we need the ability to add more flags, we could bring it in.

Also I tried to avoid making changes that would conflict with https://github.com/apache/arrow-datafusion/pull/4496 from @xudong963

# Are these changes tested?

# Are there any user-facing changes?
No

# Examples

Example running with just one file:
```shell
cargo test -p datafusion --test sqllogictests -- information_schema.slt

Running: tests/sqllogictests/test_files/information_schema.slt
Enabling information schema
[information_schema.slt] Running query: "SHOW ALL"
[information_schema.slt] Running query: "SHOW datafusion.execution.batch_size"
[information_schema.slt] Running query: "SHOW TIME ZONE"
[information_schema.slt] Running query: "SHOW TIMEZONE"
```

Example with logging:

```shell
RUST_LOG=debug cargo test -p datafusion --test sqllogictests -- information_schema.slt

[2022-12-04T12:30:03Z INFO  sqllogictests] Test directory: tests/sqllogictests/test_files
[2022-12-04T12:30:03Z INFO  sqllogictests] Running test files ["tests/sqllogictests/test_files/information_schema.slt"]
Running: tests/sqllogictests/test_files/information_schema.slt
Enabling information schema
[2022-12-04T12:30:03Z DEBUG datafusion::execution::memory_manager] Creating memory manager with initial size 11744051.2 TB
[information_schema.slt] Running query: "SHOW ALL"
[2022-12-04T12:30:03Z DEBUG sqlparser::parser] parsing expr
[2022-12-04T12:30:03Z DEBUG sqlparser::parser] prefix: Identifier(Ident { value: "name", quote_style: None })
[2022-12-04T12:30:03Z DEBUG sqlparser::parser] get_next_precedence() Comma
[2022-12-04T12:30:03Z DEBUG sqlparser::parser] 0: , 1: setting 2: FROM
....
```

### CODE REVIEW

**Comment by alamb** [comment]
cc @xudong963

File: datafusion/core/tests/sqllogictests/src/main.rs [hunk 1]
```diff
@@ -70,56 +71,96 @@ pub async fn main() -> Result<()> {
...
         "information_schema.slt" => {
-            println!("Enabling information schema");
+            info!("Enabling information schema");
             SessionContext::with_config(
                 SessionConfig::new().with_information_schema(true),
             )
         }
         _ => {
-            println!("Using default SessionContex");
+            info!("Using default SessionContex");
```
Comment (mvanschellebeeck) [inline datafusion/core/tests/sqllogictests/src/main.rs:163]: SessionContext

File: datafusion/core/tests/sqllogictests/README.md [hunk 2]
```diff
@@ -23,7 +23,21 @@ This is the Datafusion implementation of [sqllogictest](https://www.sqlite.org/s
...
+
+```shell
+RUST_LOG=debug cargo test -p datafusion --test sqllogictests
+```
+
+Run only the tests in `information_schema.slt`:
+
+```shell
+cargo test -p datafusion --test sqllogictests -- information_schema.slt
+```
```
Comment (mvanschellebeeck) [inline datafusion/core/tests/sqllogictests/README.md:40]: Could we also add a note on the substring matching ability?

**Comment by mvanschellebeeck** [comment]
nice - thanks @alamb!

**Review by xudong963 — APPROVED** [review]
This a very nice feature, we don't need to run all tests each time. Thank you @alamb

File: datafusion/core/tests/sqllogictests/src/main.rs (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/src/main.rs]: Good catch -- in https://github.com/apache/arrow-datafusion/pull/4497/commits/3be240924fe3262cd4e29b6cdf3dcc450a097d5f

File: datafusion/core/tests/sqllogictests/README.md (hunk 2 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/README.md:41]: In https://github.com/apache/arrow-datafusion/pull/4497/commits/fd98463a02895c35e74f36ff1134e4f40cab8b20

**Pull request merged**